base_domain_operator/expression.py

```diff
@@ -0,0 +1,62 @@
+# -*- coding: utf-8 -*-
+# © 2017 Therp BV <http://therp.nl>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openerp import api
+# pylint: disable=W0402
+from openerp.osv.expression import ExtendedLeaf, expression, is_operator,\
+    is_leaf
+
+
+class BaseDomainOperatorExtendedLeaf(ExtendedLeaf):
+    def __init__(self, leaf, model, join_context=None, internal=False):
+        if isinstance(leaf, ExtendedLeaf):
+            # TODO: think very long and hard if this introduces a security
+            # problem
+            internal = True
```

Comment:
Looking at for what internal is needed, it's only used for expanding the admitted operators with `inselect` and `not inselect` (ref: https://github.com/odoo/odoo/blob/a587fde991c7b3dfdffe1310d7800c210764f866/openerp/osv/expression.py#L414). I think you should leave as is without changing it.